internal/api/handlers/upsert.go

```diff
@@ -0,0 +1,43 @@
+package handlers
+
+import (
+	"awans.org/aft/internal/api/parsers"
+	"awans.org/aft/internal/bus"
+	"awans.org/aft/internal/db"
+	"awans.org/aft/internal/server/lib"
+	"github.com/json-iterator/go"
+	"net/http"
+)
+
+type UpsertHandler struct {
+	db  db.DB
+	bus *bus.EventBus
+}
+
+func (s UpsertHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) (err error) {
+	modelName, usrBody, err := unpackArgs(r)
+	if err != nil {
+		return err
+	}
+
+	tx := s.db.NewRWTx()
+	p := parsers.Parser{Tx: tx}
+
+	op, err := p.ParseUpsert(modelName, usrBody)
+	if err != nil {
+		return
+	}
+
+	s.bus.Publish(lib.ParseRequest{Request: op})
+
+	out, err := op.Apply(tx)
+	if err != nil {
+		return
+	}
+	tx.Commit()
+
+	bytes, _ := jsoniter.Marshal(&DataResponse{Data: out})
```

Comment:
should we wrap these 3 lines in a function like unpack args?
```response(DataResponse{..})```

Reply:
I went back and forth but because some are data responses and some are Batch Payloads I opted for no